core/contractsdk/cpp/XRC/xrc01/src/main/xrc01_exemple.cc

```diff
@@ -0,0 +1,414 @@
+#include <assert.h>
+#include "xchain/xchain.h"
+#include "xrc01.h"
+
+// XRC是超级链协议标准族
+// XRC_01是XRC协议家族中的第一个协议，支持在超级链中发行通过资产
+// 该协议同时支持可分割和不可分割两种资产的发行、转账、授权、授权转账、查询余额、授权关系等行为；
+
+// 通用资产协议XRC_01使用示例
+// 参数由xchain::Contract中的context提供
+class XRC01_Example {
+public:
+    /*
+     * func: 初始化智能合约
+     * @param: creator: 合约创建者
+     */
+    virtual void initialize() = 0;
+    /*
+     * func: 发行一个通用资产
+     * @param: id: 所发行的通用资产id
+     * @param: name: 所发行的通用资产名称
+     * @param: fungible: 所发行资产是否可分割
+     * @param: supply: 所发行资产的资产量，当为不可分割资产时，只能为1
+     * @param: issue_account: 初始化发行账户
+     * @param: profile_desc: 资产描述
+     */
+    virtual void issue() = 0;
+    /*
+     * func: 资产所有者将部分资产授权别人代为管理
+     * 说明: 授权的原账户为合约的发起者
+     * @param: to: 授权给的账户
+     * @param: token_id: 授权的资产id
+     * @param: amount: 授权金额
+     */
+    virtual void authorization() = 0;
+    /*
+     * func: 撤销之前授权的资产
+     * 说明: 原授权的账户才有权撤销授权, 撤销授权发起者为合约调用者
+     * @param: from: 被撤销授权的账户
+     * @param: token_id: 被撤销授权的资产id
+     * @param: amount: 被撤销授权的金额
+     */
+    virtual void withdraw_authorization() = 0;
+    /*
+     * func: 发起转账
+     * 说明: 转出账户为合约调用者
+     * @param: to: 转账接收者
+     * @param: token_id: 资产id
+     * @param: amount: 转账金额
+     */
+    virtual void transfer() = 0;
+    /*
+     * func: 被授权的账户进行代为转账
+     * @param: from: 转账发起账户
+     * @param: to: 转账接收者
+     * @param: token_id: 资产id
+     * @param: amount: 转账金额
+     */
+    virtual void authorize_transfer() = 0;
+    /*
+     * func: 查询余额
+     * @param: account: 被查询账户
+     * @param: token_id: 资产id
+     */
+    virtual void get_balance() = 0;
+    /*
+     * func: 查询账户被授权的金额
+     * @param: account: 被查询账户 
+     * @param: token_id: 资产id
+     */
+    virtual void get_authorized() = 0;
+    /*
+     * func: 查询不可分割资产所属账户
+     * @param: token_id: 资产id
+     */
+    virtual void get_owner_of() = 0;
+    /*
+     * func: 查询账户某资产的授权详情
+     * @param: account: 被查询账户 
+     * @param: token_id: 资产id
+    */
+    virtual void get_authorize_infos() = 0;
+    /*
+     * func: 查询账户某资产被授权详情
+     * @param: account: 被查询账户 
+     * @param: token_id: 资产id
+    */
+    virtual void get_authorized_infos() = 0;
+};
+
+struct XRC01_Demo : public XRC01_Example, public xchain::Contract {
+public:
+    bool str2bool(const std::string var, bool& var_bool) {
+        if (var == "1") {
+            var_bool = false;
+            return true;
+        } else if (var == "0") {
+            var_bool = true;
```

Comment:
common sense is "1" == true and "0" == false, why is it reversed here?

Reply:
Thanks, fixed